src/plugins/operators/gob_operator.py

```diff
@@ -0,0 +1,77 @@
+"""
+GOB Operator
+
+The GOB operator allows to start and control workflows in GOB
+"""
+import logging
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+from colour import Color
+
+from utils.connection import Connection
+from config.rabbitmq_config import RESULT_KEY, EXCHANGE, REQUEST_KEY
+
+
+class GOBOperator(BaseOperator):
+    ui_color = Color("lightgreen").hex
+
+    @apply_defaults
+    def __init__(self,
+                 job_name=None,
+                 step_name=None,
+                 catalogue=None,
+                 collection=None,
+                 optional=None,
+                 *args,
+                 **kwargs):
+        """
+        Initializes the GOB operator for the specific workflow
+        """
+        super().__init__(*args, **kwargs)
+
+        self.job_name = job_name
+        self.step_name = step_name
+        self.catalogue = catalogue
+        self.collection = collection
+        self.optional = optional or {}
+
+        self.connection = Connection()
+
+    def execute(self, context):
+        """
+        Execute the workflow step
+
+        Any message that is the result of a previous step in a task is available in and read from a XCom.
+
+        :param context: Airflow context
+        :return: The result of the publish of the GOB Message
+        """
+        message = context['task_instance'].xcom_pull(key=context['dag_run'].run_id)
```

Comment:
run_id has format "scheduled__YYYY-mm-DDTHH:mm:ss....". I could happen that multiple messages with the same run_id appear in the same queue. I think it will be much safer to use combination of dag_id, run_id and step_name.

Reply:
This has all been described in Jira ticket: GOB-1087
Please have a look at the ticket description.

Reply:
I still think that run_id is not good enough as a key in this case. Different tasks may be executed at the same time and this will result in multiple messages with the same run_id in the same queue.

But I admit that using the combination of dag_id, run_id and step_name is problematic because of the required knowledge of the "previous step". Let's discuss later how tackle this problem.